src/dmd/backend/os.d

```diff
@@ -972,13 +972,14 @@ else version(Solaris)
 {
 int os_critsecsize32()
 {
-    return pthread_mutex_t.sizeof;
+    assert(pthread_mutex_t.sizeof == 24);
```

Comment:
This isn't good for cross compilation (should dmd ever support it proper, which it seems to be going in the direction of).

Reply:
I tried to mimic the version OSX above. Should I also add a `version(x86_64)` check to support cross compilation?

Reply:
TBH, I'm pretty sure these functions are unused since I fixed the compiler not to depend on this library-specific value in PR #11824.